Makefile

```diff
@@ -0,0 +1,2 @@
+upgrade-submodules:
+	git submodule update --remote --init --recursive
```

Comment:
How do you remove all the checked out dirs without leaving git thinking there are unstaged changes?  Put another way, how do you undo this command?

Reply:
If you want to undo the version upgrade of the submodules, you can do:

```
git reset --hard
```

If you've modified the submodules locally, after having them upgraded (for debugging purposes perhaps) then you do:

```
git submodule foreach git reset --hard
```